scripts/lint-packages.sh

```diff
@@ -66,6 +66,52 @@ check_package_name() {
 	fi
 }
 
+check_indentation() {
+	local pkg_script="$1"
+	local line='' heredoc_terminator='' in_array=0 i=0
+	local -a issues=('' '') bad_lines=('FAILED')
+
+	# parse leading whitespace
+	while IFS=$'\n' read -r line; do
+		((i++))
+
+		# Skip this check in entirely within heredocs
+		# (see packages/ghc-libs for an example of why)
+		[[ "$line" =~ \<\<-?[\'\"]?([^\'\"]+) ]] && {
+			heredoc_terminator="${BASH_REMATCH[1]}"
+		}
+
+		(( ${#heredoc_terminator} )) && \
+		grep -qP "^\s*${heredoc_terminator}" <<< "$line" && {
+			heredoc_terminator=''
+		}
+		(( ${#heredoc_terminator} )) && continue
+
+		# check for mixed indentation
+		grep -qP '^(\t+ +| +\t+)' <<< "$line" && {
+			issues[0]='Mixed indentation'
+			bad_lines[$i]="${pkg_script}:${i}:$line"
+		}
+
+		[[ "$line" == *'=('* ]] && in_array=1
+		if (( ! in_array )); then # spaces for indentation are okay for aligning arrays
+			grep -qP '^ +' <<< "$line" && { # check for spaces as indentation
+				issues[1]='Use tabs for indentation'
+				bad_lines[$i]="${pkg_script}:${i}:$line"
+			}
+		fi
+		[[ "$line" == *')'   ]] && in_array=0
```

Comment:
```suggestion
		[[ "$line" == *')' ]] && in_array=0
```

Reply:
That was intentional to line it up with `[[ "$line" == *'=('* ]] && in_array=1`, but I forgot to change it to a single space when I moved it around.